File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.go
func searchMatrix(matrix [][]int, target int) bool {
    r, c := 0, len(matrix[0]) - 1
    
    for r < len(matrix) && c >= 0 {
        if target > matrix[r][c] {
            r += 1
        } else if target < matrix[r][c] {
            c -= 1
        } else {
            return true
        }
    }
    
    return false
}

File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        r, c = 0, len(matrix[0]) - 1
        
        while r < len(matrix) and c >= 0:
            if target > matrix[r][c]:
                r += 1
            elif target < matrix[r][c]:
                c -= 1
            else:
                return True
        
        return False